fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2669,4 +2671,31 @@ private void setOperationContextProperties(FHIROperationContext operationContext
         operationContext.setProperty(FHIROperationContext.PROPNAME_PERSISTENCE_IMPL, persistence);
         operationContext.setProperty(FHIROperationContext.PROPNAME_REQUEST_PROPERTIES, requestProperties);
     }
+
+    @Override
+    public int doReindex(FHIROperationContext operationContext, OperationOutcome.Builder operationOutcomeResult, Instant tstamp) throws Exception {
+        int result = 0;
+        // handle some retries in case of deadlock exceptions
+        final int TX_ATTEMPTS = 5;
+        int attempt = 1;
+        do {
+            FHIRTransactionHelper txn = new FHIRTransactionHelper(getTransaction());
+            txn.begin();
+            try {
+                FHIRPersistenceContext persistenceContext = null;
+                result = persistence.reindex(persistenceContext, operationOutcomeResult, tstamp);
+                attempt = TX_ATTEMPTS; // end the retry loop
+            } catch (FHIRPersistenceDataAccessException x) {
+                if (x.isTransactionRetryable() && attempt < TX_ATTEMPTS) {
+                    log.info("attempt #" + attempt + " failed, retrying transaction");
```

Comment:
Should this be a warning? I realize it's informational, and hasn't yet failed.

Reply:
Warnings might be used to generate alerts, but it hasn't yet failed so I think info is fine. Also, with the new parameter persistence code, we're no longer seeing any deadlocks during reindex.